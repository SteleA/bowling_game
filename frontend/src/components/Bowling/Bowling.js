import React, { Component } from 'react';
import { Col, Row, Grid } from 'react-bootstrap';
import ScoreForm from '../Scores/ScoreForm';
import {Scores} from '../';
import './Bowling.scss';

class Todo extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  getActiveRound() {
    return _.find(this.props.bowling.scores.slice(), {round: this.props.bowling.round}) || {};
  }

  render() {
    return (
      <Grid>
        <Row className="bowling">
          <Scores scores={this.props.bowling.scores} round={this.props.bowling.round} />
          {this.props.bowling.status === 'active' &&
            <Col sm={4} smOffset={4}>
              <ScoreForm addScore={this.props.actions.addScore} round={this.getActiveRound()} />
            </Col>
          }
        </Row>
      </Grid>
    );
  }
}

export default Todo;
