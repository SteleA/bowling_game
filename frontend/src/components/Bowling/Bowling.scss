.bowling {
  margin-top: 25px;
}
