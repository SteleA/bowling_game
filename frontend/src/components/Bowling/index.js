import Bowling from './Bowling';
export default Bowling;
