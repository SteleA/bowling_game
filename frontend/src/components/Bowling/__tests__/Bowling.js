import { shallow } from 'enzyme';  // eslint-disable-line
import { shallowToJson } from 'enzyme-to-json';  // eslint-disable-line
import React from 'react';
import Bowling from '../';

const props = {
  actions: {
    addScore: () => {},
  },
  bowling: {
    status: 'active',
    scores: [],
  },
};

describe('<Bowling />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Bowling {...props} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
