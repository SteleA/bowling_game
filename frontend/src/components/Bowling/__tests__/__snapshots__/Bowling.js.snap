// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bowling /> should render with default props 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={false}
>
  <Row
    bsClass="row"
    className="bowling"
    componentClass="div"
  >
    <scores
      scores={Array []}
    />
    <Col
      bsClass="col"
      componentClass="div"
      sm={4}
      smOffset={4}
    >
      <ScoreForm
        addScore={[Function]}
        round={Object {}}
      />
    </Col>
  </Row>
</Grid>
`;
